docs/rules/expect.md

```diff
@@ -0,0 +1,146 @@
+# expect-type/expect
+
+💼 This rule is enabled in the ✅ `recommended` config.
+
+🔧 This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).
+
+💭 This rule requires type information.
+
+<!-- end auto-generated rule header -->
+
+Enforces that types indicated in special comments match the types of code values.
+
+## Comment Types
+
+The following kinds of comments are supported:
+
+<!-- Markdownlint doesn't seem to understand the heading IDs... -->
+<!-- markdownlint-disable link-fragments -->
+
+- [`^?` (Twoslash Syntax)](#twoslash-syntax)
+- [`$ExpectError`](#expecterror)
+- [`$ExpectType`](#expecttype)
+- [`$ExpectTypeSnapshot`](#expecttypesnapshot)
+
+<!-- markdownlint-enable link-fragments -->
+
+### Twoslash Syntax (`^?`)
+
+Twoslash annotations are comment lines that start with two slashes (// ) and the ^? identifier to annotate a type below a value in code.
+For example:
+
+```ts
+declare const getTextLength: (text: string) => number;
+
+getTextLength('abc');
+// ^? number
+
+getTextLength;
+// ^? (text: string) => number
+```
+
+Mismatching the type will cause a lint report:
+
+```ts
+'abc';
+// ^? number
+```
+
+```plaintext
+Expected type to be: number, got: string
+```
+
+Multiline type annotations are also supported:
+
+```ts
+const vector = {
+  x: 3,
+  y: 4,
+};
+vector;
+// ^? const vector: {
+//        x: number;
+//        y: number;
+//    }
+```
+
+### `$ExpectError`
+
+Place this above a line of code to assert that it causes a TypeScript type error.
```

Comment:
Does it specifically have to be a type error? Can you do assertions on syntax errors or other types of errors?

Reply:
Technically it can also be anything within `ts.getPreEmitDiagnostics` -config file parsing, options- with the exception of syntax errors - as `@typescript-eslint/parser` will report a parsing error on them. So the rough description of any "type error" is pretty much accurate.